### ISSUE TITLE
Versioning of the Freva-Parts

### ISSUE THREAD

**Issue by Karinon** [FREVA-CLINT/freva-deployment#62]
This is rather something for the long run:

Considering our communication with FUB, I think, that we need to some kind of "version-orchestration" of all the freva-parts, i.E.:

"Version x of the web only works with Version x.1 - x.12 of the core."

The way how freva-deploy works right now is essentially a `give me the newest shit` and I am not sure if this is smart, especially when we want to make bigger changes and keep the other out-of-DKRZ-instances stable without forcing them to update.

**Comment by antarcticrainforest** [comment]
Very good, point! If you want to make a feasible suggestion go ahead.

**Comment by antarcticrainforest** [comment]
I was just thinking about a possible solution. I think it's best if we set versions of the parts somewhere here. Something like:

```json
{"core": "2401.0.0", "web": "2312.1.12",  "databrowser": "2402.1.1"}
```
The versions are then passed to the playbooks to pull the right containers/install the right versions.

That means that we would have to drop the modularity of the deployment. This means that you cannot for example install the web without the databrowser because a web update might assume that you have a minimum version of the databrowser set up. While this is not nice, I don't think it's too bad. We could of course add insible routines to check whether or not an update of a service - say databrowser - needs to be done before continuing with the installation of the actual service.

Also, it would be nice to have some automatism in place on the service sides that adjusts the minimum version requirement in the deployment repo and submits a PR. Otherwise, you will forget to adjust the minimum web version in the deployment when creating a new web release.

**Comment by antarcticrainforest** [comment]
I'll set up something once the current PR's are done and two more PR's that I am planning on pushing before. But overall I think this is a very good idea @Karinon
